### PR TITLE
Throttling + All History Data

### PR DESCRIPTION
**Pull request by andrew-baylis**
Introduced a throttle approach (40 calls per minute) for YahooAPI calls, and created a new historical data call which downloads dividends, splits and prices in one call rather than three.

### CODE REVIEW

File: src/YahooClient.cs [hunk 1]
```diff
@@ -5,6 +5,8 @@ public class YahooClient
     private readonly Country Country;
 
     private readonly Language Language;
+    
+    public static bool IsThrottled { get; set; }
```
Comment (ooples) [inline src/YahooClient.cs:9]: This needs to be a readonly variable

File: src/Helpers/CrumbHelper.cs [hunk 2]
```diff
@@ -1,87 +1,167 @@
...
+internal class DownloadThrottleQueueHandler : HttpClientHandler
+{
+    #region Fields
+
+    private readonly TimeSpan _maxPeriod;
+    private readonly SemaphoreSlim _throttleLoad, _throttleRate;
+
+    #endregion
+
+    public DownloadThrottleQueueHandler(int maxPerPeriod, TimeSpan maxPeriod, int maxParallel = -1)
```
Comment (ooples) [inline src/Helpers/CrumbHelper.cs:132]: Did you do testing with the 40 per minute limit to confirm this is Yahoo's hard limit?

**Review by ooples — CHANGES_REQUESTED** [review]
Just left a few minor comments to address

File: src/YahooClient.cs (hunk 1 above)
Comment (andrew-baylis) [reply src/YahooClient.cs:9]: I have it as read/write to allow throttling to be selected (or not) on program start. This allows the user to decide which mode to implement.

File: src/Helpers/CrumbHelper.cs (hunk 2 above)
Comment (andrew-baylis) [reply src/Helpers/CrumbHelper.cs:132]: I did not test the 40 per minute, but used the number based upon some comments in the community - typically 60 per minute is quoted, but there is no hard evidence to support.

**Pull request merged**